dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationHelper.java

```diff
@@ -0,0 +1,409 @@
+            throws DotSecurityException, DotDataException {
+
+        final String serviceKey = form.getServiceKey();
+        if (!UtilMethods.isSet(serviceKey)) {
+            throw new DotDataException("Required param serviceKey isn't set.");
+        }
+        final String siteId = form.getSiteId();
+        if (!UtilMethods.isSet(siteId)) {
+            throw new DotDataException("Required Param siteId isn't set.");
+        }
+        final Host host = hostAPI.find(siteId, user, false);
+        if(null == host) {
+            throw new DotDataException(String.format(" Couldn't find any site with identifier `%s` ",siteId));
+        }
+        final Optional<ServiceDescriptor> optionalServiceDescriptor = serviceIntegrationAPI
+                .getServiceDescriptor(serviceKey, user);
+        if (!optionalServiceDescriptor.isPresent()) {
+            throw new DotDataException(String.format("Unable to find a service descriptor bound to the  serviceKey `%s`. You must upload a yml descriptor.",serviceKey));
+        }
+        final Map<String, Param> params = form.getParams();
+        if(!UtilMethods.isSet(params)){
+            throw new DotDataException("Required Params aren't set.");
+        }
+        final ServiceDescriptor serviceDescriptor = optionalServiceDescriptor.get();
+        validateIncomingParams(params,serviceDescriptor, true);
+
+        final Optional<ServiceSecrets> serviceSecretsOptional = serviceIntegrationAPI
+                .getSecrets(serviceKey, host, user);
+        if (!serviceSecretsOptional.isPresent()) {
+            throw new DotDataException(String.format("Unable to find a secret for serviceKey `%s`.",serviceKey));
+        } else {
+            serviceIntegrationAPI.deleteSecret(serviceKey, params.keySet(), host, user);
+        }
+    }
+
+    /**
+     * Validate the incoming params match the params described by a serviceDescriptor yml.
+     * @param incomingParams a set of paramNames
+     * @param serviceDescriptor the service template
+     * @throws DotDataException This will give bac an exception if you send an invalid param.
+     */
+    private void validateIncomingParams(final Map<String, Param> incomingParams, final ServiceDescriptor serviceDescriptor)
+            throws DotDataException {
+            validateIncomingParams(incomingParams, serviceDescriptor, false);
+    }
+
+    /**
+     * Validate the incoming params match the params described by a serviceDescriptor yml.
+     * @param incomingParams a set of paramNames
+     * @param serviceDescriptor the service template
+     * @param skipRequiredValidation if true no required fields are validated (useful when calling it from a delete where we only need the param names)
+     * @throws DotDataException This will give bac an exception if you send an invalid param.
+     */
+    private void validateIncomingParams(final Map<String, Param> incomingParams, final ServiceDescriptor serviceDescriptor, boolean skipRequiredValidation)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'skipRequiredValidation' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)